ibc-apps/ics20-transfer/types/src/error.rs

```diff
@@ -17,86 +16,46 @@ pub enum TokenTransferError {
     ContextError(ContextError),
     /// invalid identifier: `{0}`
     InvalidIdentifier(IdentifierError),
-    /// insufficient funds: tried to send `{send_attempt}`, sender only has `{available_funds}`
-    InsufficientFunds {
-        send_attempt: String,
-        available_funds: String,
-    },
-    /// destination channel not found in the counterparty of port_id `{port_id}` and channel_id `{channel_id}`
-    DestinationChannelNotFound {
-        port_id: PortId,
-        channel_id: ChannelId,
-    },
-    /// base denomination is empty
-    EmptyBaseDenom,
-    /// invalid prot id n trace at position: `{pos}`, validation error: `{validation_error}`
-    InvalidTracePortId {
-        pos: u64,
-        validation_error: IdentifierError,
-    },
-    /// invalid channel id in trace at position: `{pos}`, validation error: `{validation_error}`
-    InvalidTraceChannelId {
-        pos: u64,
-        validation_error: IdentifierError,
-    },
-    /// malformed trace: `{0}`
-    MalformedTrace(String),
-    /// trace length must be even but got: `{len}`
-    InvalidTraceLength { len: u64 },
+    /// invalid trace: `{0}`
+    InvalidTrace(String),
     /// invalid amount: `{0}`
     InvalidAmount(FromDecStrErr),
     /// invalid token
     InvalidToken,
-    /// expected `{expect_order}` channel, got `{got_order}`
-    ChannelNotUnordered {
-        expect_order: Order,
-        got_order: Order,
+    /// invalid coin: `{actual}`
+    InvalidCoin { actual: String },
```

Comment:
Should get the naming consistent? Either leave with variants as **Token or **Coin ?

Reply:
The semantics around Token and Coin are used inconsistently throughout the codebase more generally. Conceptually, they might be interchangeable, but the the `InvalidCoin` error variant is used in a `FromStr for Coin` implementation, so it's a bit strange to change this to variant to `InvalidToken` when it's called `Coin` in the surrounding context where it is used. 

In a similar vein with changing all error variants to use `Coin` instead of `Token`, `Token` definitely is more prevalently used than `Coin` in the codebase, as well as the application being called `TokenTransfer`. So I'm not really sure if changing the names of these variants to be consistently `Token` or `Coin` wouldn't be confusing.

Reply:
You are right, and your view makes a lot of sense. Let's keep it as it is.